test/e2e/set-manifest-flags.ts

```diff
@@ -8,23 +9,82 @@ function parseIntOrUndefined(value: string | undefined): number | undefined {
   return value ? parseInt(value, 10) : undefined;
 }
 
-// Grab the tracesSampleRate from the git message if it's set
-function getTracesSampleRateFromGitMessage(): number | undefined {
+/**
+ * Search a string for `flags = {...}` and return ManifestFlags if it exists
+ *
+ * @param str - The string to search
+ * @param errorType - The type of error to log if parsing fails
+ * @returns The ManifestFlags object if valid, otherwise undefined
+ */
+function regexSearchForFlags(
+  str: string,
+  errorType: string,
+): ManifestFlags | undefined {
+  // Search str for `flags = {...}`
+  const flagsMatch = str.match(/flags\s*=\s*(\{.*\})/u);
```

Comment:
This will also match e.g. `unrelatedflags = {...}` (maybe that's fine?)

Reply:
Good point, but yes, that does seem fine